vm/vm.go

```diff
@@ -433,6 +433,34 @@ func (vm *VM) Initialize(
 	webSocketServer, pubsubServer := rpc.NewWebSocketServer(vm, vm.config.StreamingBacklogSize)
 	vm.webSocketServer = webSocketServer
 	vm.handlers[rpc.WebSocketEndpoint] = pubsubServer
+
+	has, err = vm.HasLastProcessed()
+	if err != nil {
+		snowCtx.Log.Error("could not determine if have last processed")
+		return err
+	}
+	if has {
+		lastProcessedHeight, err := vm.GetLastProcessedHeight()
+		if err != nil {
+			snowCtx.Log.Error("could not get last processed height", zap.Error(err))
+			return err
+		}
+
+		for height := lastProcessedHeight; height <= vm.lastAccepted.Height(); height++ {
+			var blk *chain.StatelessBlock
+			if height == vm.lastAccepted.Height() {
+				blk = vm.lastAccepted
+			} else {
+				blk, err = vm.GetDiskBlock(ctx, height)
+				if err != nil {
+					vm.snowCtx.Log.Info("could not find block on-disk", zap.Uint64("height", height))
+					return err
+				}
+			}
+			vm.processAcceptedBlock(blk)
+		}
+	}
+
```

Comment:
This `Initialize` function is already huge. Could we move this into its own function since it is pretty well-defined?

Reply:
Moved it to the new `restoreLastAcceptedBlock` that is called at the end of the initialization of the `VM`.